### PR TITLE
CMakeList - BUILD_DEV Option Added

### PR DESCRIPTION
**Pull request by kiritigowda**
Build Dev set a default
set `-D BUILD_DEV=OFF` to build runtime only package

### CODE REVIEW

**Comment by LakshmiKumar23** [comment]
@kiritigowda The ```mvdeploy_api.h``` seems to included even if ```BUILD_DEV=OFF```. Everything else looks okay.

```svcbuild@Legolas:~/test/MIVisionX/build-nodev/mivisionx-2.4.0-Linux/opt/rocm/include/mivisionx$ ls mvdeploy_api.h```

File: amd_openvx_extensions/amd_rpp/CMakeLists.txt [hunk 1]
```diff
@@ -125,52 +131,53 @@ list(APPEND SOURCES
...
     add_definitions(-DENABLE_OPENCL=${ENABLE_OPENCL} -DENABLE_HIP=${ENABLE_HIP} -DRPP_BACKEND_HIP=1 -D__HIP_PLATFORM_AMD__)
     include_directories(${ROCM_PATH}/include)
     link_directories(${HIP_PATH}/lib)
+    set(LINK_LIBRARY_LIST ${LINK_LIBRARY_LIST} hip::host)
     add_library(vx_rpp SHARED ${SOURCES})
     set_target_properties(openvx PROPERTIES LINKER_LANGUAGE CXX)
     set_target_properties(openvx PROPERTIES POSITION_INDEPENDENT_CODE ON)
     # TBD -- CHECK to FIX
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DGPU_SUPPORT=1 -DRPP_BACKEND_HIP=1") # Currently used by rpp. To be removed.
-    target_link_libraries(vx_rpp amd_rpp boost_filesystem boost_system hip::host)
```
Comment (kiritigowda) [inline amd_openvx_extensions/amd_rpp/CMakeLists.txt:151]: @LakshmiKumar23 had boost required but never checked for library, had to fix in this pr.

Comment (LakshmiKumar23) [reply amd_openvx_extensions/amd_rpp/CMakeLists.txt:151]: > @kiritigowda The `mvdeploy_api.h` seems to included even if `BUILD_DEV=OFF`. Also seeing the mv_deploy scripts in model compiler
> 
> ```
> svcbuild@Legolas:~/test/MIVisionX/build-nodev/mivisionx-2.4.0-Linux/opt/rocm/include/mivisionx$ ls 
> mvdeploy_api.h
> svcbuild@Legolas:~/test/MIVisionX/build-nodev/mivisionx-2.4.0-Linux/opt/rocm/libexec/mivisionx/model_compiler/mv_deploy$ ls
> mv_extras_postproc.cpp  mv_extras_postproc.h
> ```

@kiritigowda can you look into this also?

**Review by LakshmiKumar23 — APPROVED** [review]
@kiritigowda Verified after last commit. Installs correctly

**Pull request merged**